hadoop-hdds/common/src/main/resources/ozone-default.xml

```diff
@@ -3275,6 +3275,24 @@
     </description>
   </property>
 
+  <property>
+    <name>ozone.client.ec.grpc.retries.enabled</name>
+    <value>true</value>
+    <tag>CLIENT</tag>
+    <description>
+      To enable Grpc client retries for EC.
+    </description>
+  </property>
+
+  <property>
```

Comment:
What is the time out it declares DEADLINE EXCEEDED?
total wait before failing would be: 5 * that_timeout right ?
Should we just set 3? I don't know the reason, but if application just blocked for that much time to get response, then app may behave something wrong? I don't know. 
Can you let me know the total possible wait time?

Reply:
The client timeout is specified by `ozone.client.read.timeout` which is by default `30s`, After which GRPC throws `DEADLINE_EXCEEDED`. Under heavy load, If we assume we get the same error for the 5 retries, then the wait time would be `5 * 30s = 150s`. 

One solution is that we can make the retry count to 3 (which also includes transient retries grpc tries on its own) so we don't try as much because we don't want to burden the DN that much as well. Or we can also try [throttling the retries](https://github.com/grpc/proposal/blob/master/A6-client-retries.md#throttling-retry-attempts-and-hedged-rpcs)

Reply:
Thanks for the details @aswinshakil. Please file a JIRA for metrics or some way we should let people know about these retries happening. I am ok with 3. BTW, I don't think we will burden the DN because, calls may not really reach DN right?

Reply:
Sure will file a JIRA and update the retry count. Initially, I kept the retry count to 3, thought it might be too small. But makes sense to change it back to 3. 

If the call doesn't even reach the DN, then it won't throw `DEADLINE_EXCEEDED` and rather throw `UNAVAILABLE`, so we should be fine in that case.

Reply:
Jira for GRPC metrics [HDDS-7530](https://issues.apache.org/jira/browse/HDDS-7530)